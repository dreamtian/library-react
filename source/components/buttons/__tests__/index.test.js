
import { Component } from 'react';
import { render, mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Button from '../index';
// import { wrap } from 'module';


describe('Button', () => {
  it('renders correctly', () => {
    const wrapper = render(
      <Button></Button>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('two CN char join space', () => {
    const wrapper = render(
      <Button>嘻嘻</Button>
    );
    expect(wrapper.text()).toBe('嘻 嘻');
  })

  it('should response click event to change text', () => {
    class DefaultButton extends Component {
      state = {
        text: 'init'
      }
      enterDisable = () => {
        this.setState({
          text: 'changed'
        })
      }
      render() {
        return <Button onClick={this.enterDisable} disabled={this.state.disabled}>{this.state.text}</Button>
      }
    }
    const wrapper = mount(
      <DefaultButton />
    );
    expect(wrapper.find('button').text()).toBe('init');
    wrapper.simulate('click');
    expect(wrapper.find('button').text()).toBe('changed');
  })
});