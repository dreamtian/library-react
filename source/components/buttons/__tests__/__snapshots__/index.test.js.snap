// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly 1`] = `
<button
  class="test-btn"
  type="common"
/>
`;

exports[`Button should response click event to change text 1`] = `
<DefaultButton>
  <Button
    disabled={false}
    nbButton={false}
    onClick={[Function]}
    prefixCls="test-btn"
    size="common"
    type="common"
  >
    <button
      className="test-btn"
      disabled={false}
      onClick={[Function]}
      size="common"
      type="common"
    >
      <span
        key=".0"
      >
        changed
      </span>
    </button>
  </Button>
</DefaultButton>
`;
